packages/oui-datagrid/src/table.controller.js

```diff
@@ -63,33 +65,80 @@ export default class {
         this.canClickOnRow = this.$attrs.onRowClick;
     }
 
-    init () {
-        // Local data
-        if (this.rows) {
-            this.$scope.$watchCollection("tableCtrl.rows", () => {
-                this.filteredRows = this.rows;
-                this.sortedRows = this.rows;
-
-                this.updatePageMeta({
-                    currentOffset: 0,
-                    pageCount: Math.ceil(this.rows.length / this._pageSize),
-                    totalCount: this.rows.length
-                });
+    $postLink () {
+        this.$transclude((clone) => {
+            const columnElements = DatagridController.filterElements(clone, "column");
 
-                this.changePage()
-                    .catch(this.handleError.bind(this));
-            });
-        } else
+            const builtColumns = this.ouiTableColumnBuilder.build(columnElements, this.$scope);
+            this.columns = builtColumns.columns;
+            this.currentSorting = builtColumns.currentSorting;
+
+            // Once columns has been processed,
+            // we can proceed with the first loading
+            this.initPage();
+
+            const paginationElement = this.$element.find("pagination");
+            if (paginationElement.length) {
+                this.setPaginationTemplate(paginationElement.html());
+            }
+        });
+    }
+
+    $doCheck () {
+        if (!angular.equals(this.previousRows, this.rows)) {
+            this.previousRows = angular.copy(this.rows);
+
+            if (this.rows) {
+                this.updateRows();
+            }
+        }
+    }
+
+    static filterElements (elements, tagName) {
```

Comment:
(optional) Maybe sort method if we use static ones... ?

Reply:
Lets talk about it together and I'll make changes based on it :D